### PR TITLE
[release-1.25] OCPBUGS-23951: Update various dependencies to fix outstanding CVEs

### PR DESCRIPTION
**Pull request by kwilczynski**
#### What type of PR is this?

/kind dependency-change
/assign kwilczynski

#### What this PR does / why we need it:

Update the following dependencies to fix outstanding CVEs:

- go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp [GHSA-rcjv-mgp8-qvmr](https://osv.dev/vulnerability/GHSA-rcjv-mgp8-qvmr)
- go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc  [GHSA-8pgv-569h-w5rw](https://osv.dev/vulnerability/GHSA-8pgv-569h-w5rw)
- github.com/go-jose/go-jose/v3 [GHSA-2c7c-3mj9-8fqh](https://osv.dev/vulnerability/GHSA-2c7c-3mj9-8fqh) / [GO-2023-2334](https://pkg.go.dev/vuln/GO-2023-2334)

Related:

- https://github.com/cri-o/cri-o/pull/7526
- https://github.com/cri-o/cri-o/pull/7504
- https://github.com/cri-o/cri-o/pull/7489
- https://github.com/cri-o/cri-o/pull/7527
- https://github.com/cri-o/cri-o/pull/7537
- https://github.com/cri-o/cri-o/pull/7545
- [cc956e6: Move to use stats handlers over deprecated interceptors](https://github.com/cri-o/cri-o/pull/7526/commits/cc956e6ba68f1dc0eedcf72bd29d3f6c2d5da801)

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -17,15 +17,15 @@ require (
...
+	github.com/containers/common v0.51.0
 	github.com/containers/conmon v2.0.20+incompatible
 	github.com/containers/conmon-rs v0.4.0
-	github.com/containers/image/v5 v5.25.0
+	github.com/containers/image/v5 v5.24.0
 	github.com/containers/kubensmnt v1.2.0
 	github.com/containers/ocicrypt v1.1.7
-	github.com/containers/podman/v4 v4.5.2-0.20230706090613-813f1b53bf39
-	github.com/containers/storage v1.46.1
+	github.com/containers/podman/v4 v4.4.0-rc2
```
Comment (haircommander) [inline go.mod:27]: maybe 4.4.1?

File: internal/oci/oci_test.go.orig [hunk 2]
```diff
@@ -0,0 +1,356 @@
+package oci_test
```
Comment (haircommander) [inline internal/oci/oci_test.go.orig:1]: this seems unintentional

File: go.mod (hunk 1 above)
Comment (TomSweeneyRedHat) [reply go.mod:27]: That will bump c/storage to 1.45.6, is that OK?

File: internal/oci/oci_test.go.orig (hunk 2 above)
Comment (kwilczynski) [reply internal/oci/oci_test.go.orig:1]: @haircommander, good catch!

Sorry about this!

Comment (kwilczynski) [reply internal/oci/oci_test.go.orig]: Done.

Completed in https://github.com/cri-o/cri-o/commit/4d9585de0d2643c789a7c19ef2fbe66af80d166b.

File: go.mod (hunk 1 above)
Comment (kwilczynski) [reply go.mod:27]: Related to:

- https://github.com/containers/podman/pull/20881

Comment (kwilczynski) [reply go.mod:27]: @TomSweeneyRedHat, that would be fine. We will be updating containers/storage to **v1.45.6** anyway.

File: go.mod [hunk 3]
```diff
@@ -17,15 +17,15 @@ require (
...
 	github.com/containernetworking/cni v1.1.2
 	github.com/containernetworking/plugins v1.2.0
-	github.com/containers/buildah v1.30.0
-	github.com/containers/common v0.52.0
+	github.com/containers/buildah v1.29.0
+	github.com/containers/common v0.51.0
 	github.com/containers/conmon v2.0.20+incompatible
 	github.com/containers/conmon-rs v0.4.0
-	github.com/containers/image/v5 v5.25.0
+	github.com/containers/image/v5 v5.24.0
```
Comment (haircommander) [inline go.mod:24]: /hold

didn't we decide to stay here on this branch?

Comment (kwilczynski) [reply go.mod:24]: @haircommander, since we want to keep containers/storage at 1.46.x, I will clean some of the extra commits up.

Comment (kwilczynski) [reply go.mod]: Done.

**Review by mrunalp — APPROVED** [review]
/lgtm

**Pull request merged**